vllm/v1/worker/gpu_model_runner.py

```diff
@@ -421,6 +432,81 @@ def _prepare_inputs(self, scheduler_output: "SchedulerOutput"):
         # where M is the max_model_len.
         token_indices = (positions_np +
                          req_indices * self.input_batch.token_ids_cpu.shape[1])
+
+        use_spec_decode = len(all_spec_token_ids) > 0
+        if use_spec_decode:
+            # Currently, we assume all speculated tokens are verified.
+            # We don't support partial verification.
```

Comment:
What does this mean? What's the blocker of this?

Reply:
I think I should move this comment to `scheduler`'s `update_from_output`.
Currently, we clear all spec tokens in scheduler `update_from_output`, which means all speculated tokens are verified in the last generation step. If we only verify partial spec tokens, we should not clear all of them.